django_ontruck/consumer.py

```diff
@@ -0,0 +1,54 @@
+from abc import ABCMeta, abstractmethod
+from celery import bootsteps
+from kombu import Consumer, Exchange, Queue
+from django.conf import settings
+
+
+class ConsumerBase(bootsteps.ConsumerStep):
+    __metaclass__ = ABCMeta
+
+    @property
+    @abstractmethod
+    def queue_name(self):
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def exchange_name(self):
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def routing_key(self):
+        raise NotImplementedError()
+
+    @property
+    def tag_prefix(self):
+        return None
+
+    @abstractmethod
+    def perform(self, body, message):
+        raise NotImplementedError('missing perform')
+
+    def get_consumers(self, channel):
+        queue = Queue(self.queue_name, Exchange(self.exchange_name), routing_key=self.routing_key)
+        return [
+            Consumer(
+                channel,
+                queues=[queue],
+                callbacks=[self.handle_message],
+                accept=['json', 'application/json'],
+                tag_prefix=self.tag_prefix),
+        ]
+
+    def handle_message(self, body, message):
+        try:
+            self.perform(body, message)
```

Comment:
Thinking about the guidelines in the `perform` part maybe in future iterations we could give some code block just to define `serializer` and `use case` to control almost all the structure

Reply:
We can follow the DRF structure and do something like:
```python
class MyConsumer(ConsumerBase):
    serializer_class = MySerializer
    use_case_class = MyUseCase
```

Reply:
Yes, our django-ontruck views follows that structure too
https://github.com/ontruck/django-ontruck/blob/master/django_ontruck/views.py#L41